Flagsmith.FlagsmithClient/FlagsmithClient.cs

```diff
@@ -103,7 +103,7 @@ public class FlagsmithClient : IFlagsmithClient
             if (EnableClientSideEvaluation)
             {
                 _pollingManager = new PollingManager(GetAndUpdateEnvironmentFromApi, EnvironmentRefreshIntervalSeconds);
-                _ = _pollingManager.StartPoll();
+                _pollingManager.StartPoll().GetAwaiter().GetResult();
```

Comment:
This looks a bit ugly - what's the reasoning here?

Reply:
May I ask why it looks a bit ugly? The reasoning is just to accomplish the issue objective:

> Ensures environment is retrieved on start polling by executing the StartPoll() method synchronously

Why use this particular way to make the call async? As per some tech docs I found, like this one https://devblogs.microsoft.com/dotnet/configureawait-faq/, this is the way the compiler would interpret an `await task ` code, and also, it would handle the error bubbling up an internal error happening in an internal node of a nested call, so we don't need to change our error handlers to use an error stacker as mentioned here https://gsferreira.com/archive/2020/08/avoid-getawaiter-getresult-at-all-cost/.

Making an async method work synchronously is strongly unrecommended in general, but this seems to be one of the few cases where it is needed since we want to call it in a constructor method, which by definition should execute synchronously since it should complete any initialization of the instance before any other action with the instance.